### PR TITLE
[arrow] add plasma support for non Windows platforms

### PR DESCRIPTION
**Pull request by fran6co**
**Describe the pull request**

Add a way to enable plasma

### CODE REVIEW

File: ports/arrow/vcpkg.json [hunk 1]
```diff
@@ -76,6 +77,9 @@
     "parquet": {
       "description": "Parquet support"
     },
+    "plasma": {
+      "description": "Plasma support"
```
Comment (autoantwort) [inline ports/arrow/vcpkg.json:81]:
```suggestion
      "description": "Plasma support",
      "supports": "!windows"
```

File: ports/arrow/portfile.cmake [hunk 2]
```diff
@@ -32,9 +32,14 @@ vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
...
         parquet     ARROW_PARQUET
         parquet     PARQUET_REQUIRE_ENCRYPTION
+        plasma      ARROW_PLASMA
         s3          ARROW_S3
 )
 
+if (VCPKG_TARGET_IS_WINDOWS AND ("plasma" IN_LIST FEATURES))
+    message(FATAL_ERROR "Unsupported plasma on Windows")
+endif ()
+
```
Comment (autoantwort) [inline ports/arrow/portfile.cmake:42]:
```suggestion
```

**Review by autoantwort — CHANGES_REQUESTED** [review]

**Comment by LilyWangLL** [comment]
Thanks for posting this PR. I test arrow[plasma] on Linux, and it build with the following error:
```
There should be no bin\ directory in a static build, but /home/vlilywang/Lily/vcpkg/packages/arrow_x64-linux/bin is present.
There should be no debug\bin\ directory in a static build, but /home/vlilywang/Lily/vcpkg/packages/arrow_x64-linux/debug/bin is present.
If the creation of bin\ and/or debug\bin\ cannot be disabled, use this in the portfile to remove them

    if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
        file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/bin" "${CURRENT_PACKAGES_DIR}/debug/bin")
    endif()
```
I found this exe file under the directory ${CURRENT_PACKAGES_DIR}/bin, you can use `vcpkg_copy_tools(TOOL_NAMES plasma-store-server AUTO_CLEAN)` to fix it.
image

**Comment by fran6co** [comment]
> Thanks for posting this PR. I test arrow[plasma] on Linux, and it build with the following error:
> 
> ```
> There should be no bin\ directory in a static build, but /home/vlilywang/Lily/vcpkg/packages/arrow_x64-linux/bin is present.
> There should be no debug\bin\ directory in a static build, but /home/vlilywang/Lily/vcpkg/packages/arrow_x64-linux/debug/bin is present.
> If the creation of bin\ and/or debug\bin\ cannot be disabled, use this in the portfile to remove them
> 
>     if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
>         file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/bin" "${CURRENT_PACKAGES_DIR}/debug/bin")
>     endif()
> ```
> 
> I found this exe file under the directory ${CURRENT_PACKAGES_DIR}/bin, you can use `vcpkg_copy_tools(TOOL_NAMES plasma-store-server AUTO_CLEAN)` to fix it. image

Done

**Review by LilyWangLL — APPROVED** [review]

**Pull request merged**